r/src/main/scala/org/apache/zeppelin/rinterpreter/RInterpreter.scala

```diff
@@ -141,8 +142,15 @@ object RInterpreter {
   }
 
   def dataURI(file : String, mime : String) : String = {
-    val data: String = Source.fromFile(file).getLines().mkString("\n")
-    s"""data:${mime};base64,""" + StringUtils.newStringUtf8(Base64.encodeBase64(data.getBytes(), false))
+    val fp = new File(file)
+    val fdata = new Array[Byte](fp.length().toInt)
+    val fin = new BufferedInputStream(new FileInputStream(fp))
+    try {
+      fin.read(fdata)
```

Comment:
yes... here read `fp.length()` bytes from the file, so it read the whole file...